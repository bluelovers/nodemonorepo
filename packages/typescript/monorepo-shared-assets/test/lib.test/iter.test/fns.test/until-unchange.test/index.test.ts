import {iter} from '../../../../../index'
const {untilUnchange} = iter.fns
const init = 65536
const callback = (x: number) => x >> 1
const equal = (a: number, b: number) => a === b

it('default comparator', () => {
  expect(Array.from(untilUnchange(init, callback))).toMatchSnapshot()
})

it('custom comparator', () => {
  expect(Array.from(untilUnchange(init, callback, equal))).toMatchSnapshot()
})
