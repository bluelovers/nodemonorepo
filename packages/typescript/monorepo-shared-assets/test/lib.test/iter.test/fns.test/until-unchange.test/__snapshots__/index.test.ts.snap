// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom comparator 1`] = `
Array [
  65536,
  32768,
  16384,
  8192,
  4096,
  2048,
  1024,
  512,
  256,
  128,
  64,
  32,
  16,
  8,
  4,
  2,
  1,
  0,
]
`;

exports[`default comparator 1`] = `
Array [
  65536,
  32768,
  16384,
  8192,
  4096,
  2048,
  1024,
  512,
  256,
  128,
  64,
  32,
  16,
  8,
  4,
  2,
  1,
  0,
]
`;
